filecopy/operations/services/notification/client.py

```diff
@@ -0,0 +1,104 @@
+from typing import List
+from typing import Optional
+
+from operations.models import Node
+from operations.models import NodeList
+from operations.services.notification.models import InvolvementType
+from operations.services.notification.models import Location
+from operations.services.notification.models import NotificationType
+from operations.services.notification.models import PipelineAction
+from operations.services.notification.models import PipelineNotification
+from operations.services.notification.models import PipelineStatus
+from operations.services.notification.models import Target
+from requests import Session
+
+logger = logging.getLogger(__name__)
+
+
+class NotificationServiceClient:
+    """Client for sending notifications into Notification Service."""
+
+    def __init__(
+        self,
+        endpoint: str,
+        include_nodes: NodeList,
+        source_folder: Node,
+        destination_folder: Optional[Node],
+        project_code: str,
+        pipeline_action: PipelineAction,
+        pipeline_status: PipelineStatus,
+        operator: str,
+        notification_type: NotificationType,
+    ) -> None:
+        self.endpoint = f'{endpoint}/v1/all/notifications/'
+        self.include_nodes = include_nodes
+        self.source_folder = source_folder
+        self.destination_folder = destination_folder
+        self.project_code = project_code
+        self.pipeline_action = pipeline_action
+        self.pipeline_status = pipeline_status
+        self.operator = operator
+        self.notification_type = notification_type
+        self.client = Session()
+
+    def set_status(self, status: str) -> None:
+        self.pipeline_status = status
+
+    def set_location(self, entity: Node) -> Location:
+        return Location(id=entity.id, path=str(entity.display_path), zone=entity.zone).dict()
+
+    def set_targets(self) -> List[dict]:
+        targets = []
+        for node in self.include_nodes:
+            file_node = self.include_nodes[node]
+            targets.append(Target(id=file_node.id, name=file_node.name, type=file_node.entity_type).dict())
+        return targets
+
+    def get_priority(self) -> dict:
+        involvers = {InvolvementType.INITIATOR: self.operator}
+        owner = self.source_folder.display_path.parts[0]
+        receiver = self.destination_folder.display_path.parts[0] if self.destination_folder else None
+        if owner != self.operator:
+            involvers[InvolvementType.OWNER] = owner
+        elif receiver not in [owner, self.operator]:
+            involvers[InvolvementType.RECEIVER] = receiver
+        return involvers
+
+    def send_notifications(self) -> None:
+        source_folder = self.set_location(self.source_folder)
+        targets_entity = self.set_targets()
+        involvers = self.get_priority()
+        payload = []
+        destination_folder = self.set_location(self.destination_folder) if self.destination_folder else None
+        for item in involvers:
```

Comment:
Here also you can also improve readability: 
```suggestion
        for involvement, username in involvers.items():
```